internal/testrunner/runners/system/runner.go

```diff
@@ -727,6 +727,11 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 		result.FailureMsg = message
 	}
 
+	// Check transforms if present
+	if err := r.checkTransforms(config, pkgManifest, ds, dataStream); err != nil {
+		return result.WithError(err)
+	}
```

Comment:
Yeah, I think it wouldn't be needed because the source data is already deleted. The transform should be deleted by Fleet, but I think it is not doing it.